### PR TITLE
Fix handling of LNURLw response types

### LINKED ISSUE #655: LNURL - Withdraw error message
The LNURL-Withdraw is working but this message is displayed.
In the pictures are the tests from lightning.gifts 101 sats and opentip.io 14 sats

image
image

[c-breez.logs (11).zip](https://github.com/breez/c-breez/files/12818919/c-breez.logs.11.zip)

### PR DESCRIPTION
**Pull request by ok300**
Fixes #655

### CODE REVIEW

**Review by erdemyerebasmaz — APPROVED** [review]

File: lib/routes/lnurl/withdraw/lnurl_withdraw_dialog.dart [hunk 1]
```diff
@@ -148,12 +148,12 @@ class _LNURLWithdrawDialogState extends State<LNURLWithdrawDialog> with SingleTi
         amountSats: widget.amountSats,
         description: description,
       );
-      if (resp is sdk.LnUrlWithdrawSuccessData) {
-        final paymentHash = (resp as sdk.LnUrlWithdrawSuccessData).invoice.paymentHash;
+      if (resp is sdk.LnUrlWithdrawResult_Ok) {
+        final paymentHash = (resp.data as sdk.LnUrlWithdrawSuccessData).invoice.paymentHash;
```
Comment (erdemyerebasmaz) [inline lib/routes/lnurl/withdraw/lnurl_withdraw_dialog.dart:152]: We don't need to cast the type here as it's inferred by the if condition.

```suggestion
        final paymentHash = resp.data.invoice.paymentHash;
```

File: lib/routes/lnurl/withdraw/lnurl_withdraw_dialog.dart [hunk 2]
```diff
@@ -148,12 +148,12 @@ class _LNURLWithdrawDialogState extends State<LNURLWithdrawDialog> with SingleTi
...
         _log.v("LNURL withdraw success for $paymentHash");
         return const LNURLPageResult(protocol: LnUrlProtocol.Withdraw);
-      } else if (resp is sdk.LnUrlErrorData) {
-        final reason = (resp as sdk.LnUrlErrorData).reason;
+      } else if (resp is sdk.LnUrlWithdrawResult_ErrorStatus) {
+        final reason = (resp.data as sdk.LnUrlErrorData).reason;
```
Comment (erdemyerebasmaz) [inline lib/routes/lnurl/withdraw/lnurl_withdraw_dialog.dart:156]: Same here.
```suggestion
        final reason = resp.data.reason;
```

**Review by erdemyerebasmaz — APPROVED** [review]
Approved with 2 NIT comments

**Comment by ok300** [comment]
Thanks @erdemyerebasmaz !

**Pull request merged**